portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,19 +38,51 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> list = new ArrayList<>();    
+  String commentsJson = new String();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println(list);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    //Builds list of Comment Objects
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String textInput = (String) entity.getProperty("comments");
+      
+      Comment c = new Comment(id, name, textInput, timestamp);
+      comments.add(c);
+    }
+    //converts comments list to JSON
+    Gson gson = new Gson();
+    commentsJson = gson.toJson(comments);
```

Comment:
Since `gson` is only used once, consider inlining it:

```suggestion
    commentsJson = new Gson().toJson(comments);
```